woocommerce/woocommerce-gutenberg-products-block
# Use Server Side Rendering for Product Category List block

This PR updates the Product Category List Block to use SSR instead of client side rendering in order to avoid passing a full list of terms via the `wcSettings` implementation on most page loads. As a result of this, `get_terms` is only invoked during the **rendering of the block**.

There is one gotcha however which I was unable to rectify, and perhaps @nerrad can confirm if this is a bug in SSR in Gutenberg. You'll notice I added `deprecation` to ensure existing blocks do not invalidate in the editor - this all works fine editor side, but if you view an existing block on the frontend, the saved attributes are not respected and defaults get used. This is only fixed by saving. @nerrad ideas on this one? 

Closes #1005

### Screenshots

![Screenshot 2019-10-07 at 17 25 16](https://user-images.githubusercontent.com/90977/66330016-75270a80-e927-11e9-9674-8c123c608581.png)

### How to test the changes in this Pull Request:

1. Create some product category list blocks (dropdown and list) before using this PR
2. Use PR. Confirm they transform/save correctly.
3. Confirm dropdown + list appearance is unchanged on frontend.
4. Test the dropdown navigates to the correct page on click.
5. Confirm hierarchy/count toggles update the list correctly.



### Changelog

> Use Server Side Rendering for Product Category List block to remove the need to pass large amounts of data around when not needed.
